cmd/infracost/generate.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/infracost/infracost/internal/config/template"
+	"github.com/infracost/infracost/internal/ui"
+)
+
+type generateConfigCommand struct {
+	projectPath  string
+	templatePath string
+	outFile      string
+}
+
+func newGenerateConfigCommand() *cobra.Command {
+	var gen generateConfigCommand
+
+	cmd := &cobra.Command{
+		Use:   "config",
+		Short: "Generate Infracost YML file from a template file",
+		Long:  "Generate Infracost YML file from a template file",
+		Example: `
+      infracost generate config --project-path . --template-path infracost.yml.tmpl
+      `,
+		ValidArgs: []string{"--", "-"},
+		RunE:      gen.run,
+	}
+
+	cmd.Flags().StringVar(&gen.projectPath, "project-path", ".", "Path to the Terraform project you want to run the template file on")
```

Comment:
Not sure if `project` is the right term here. We just use `--path` elsewhere to refer to the repository path, but understand we might want to make this more explicit. Any ideas @alikhajeh1?

Reply:
@aliscott and @hugorut yeah `project` is confusing here as we use that term to mean another thing in Infracost.
Maybe `--repo-path` with `Path to the Terraform repo or directory you want to run the template file on`